src/mmw/apps/geoprocessing_api/calcs.py

```diff
@@ -59,12 +59,21 @@ def stream_data(results, geojson):
     Returns a dictionary to append to outgoing JSON for analysis results.
     """
 
+    NULL_SLOPE = -9998.0
+
     sql = '''
-        SELECT sum(lengthkm) as lengthkm, stream_order
+        SELECT sum(lengthkm) as lengthkm,
+               stream_order,
+               sum(case when NULLIF(slope, {NULL_SLOPE}) is NULL
+                       then
+                           NULL
+                       else
+                           slope * lengthkm
```

Comment:
`total_weighted_slope` is now `slope * lengthkm` here. Then on https://github.com/WikiWatershed/model-my-watershed/pull/2551/files#diff-15b3e3581e8f864be02ed1013a02b466R685 we do:

```
data.pluck('total_weighted_slope').sum() / totalLength
```

That seems right to me but cc @rajadain for another double check.

Q: is `data.pluck` tossing out any null values it sees in before the `sum` op?

Reply:
- The grand total is `sum(slope_i * lengthkm_i) for i=0; i<all streams`. This is equivalent to breaking out the sum for each stream order, and then adding them together: 
```
  sum(slope_i * lengthkm_i) for i=0; i < order 1 streams +
  sum(slope_i * lengthkm_i) for i=0; i< order 2 streams +
  ...
  sum(slope_i * lengthkm_i) for i=0; i< order N streams +
```

- `data.pluck` doesn't toss out null values.
```js
> _.pluck([{ a: 0 }, { a: null }, { a: 123 }], 'a');
(3) [0, null, 123]
```